src/main/scala/org/clulab/habitus/utils/ContextExtractor.scala

```diff
@@ -0,0 +1,260 @@
+                          mostFreqLocation1Sent:Seq[MostFreqEntity],
+                          mostFreqLocationOverall:Seq[MostFreqEntity],
+                          mostFreqDate0Sent:Seq[MostFreqEntity],
+                          mostFreqDate1Sent:Seq[MostFreqEntity],
+                          mostFreqDateOverall:Seq[MostFreqEntity],
+                          mostFreqCrop0Sent:Seq[MostFreqEntity],
+                          mostFreqCrop1Sent:Seq[MostFreqEntity],
+                          mostFreqCropOverall:Seq[MostFreqEntity],
+                          mostFreqFertilizer0Sent:Seq[MostFreqEntity],
+                          mostFreqFertilizer1Sent:Seq[MostFreqEntity],
+                          mostFreqFertilizerOverall:Seq[MostFreqEntity]
+                         ): Unit= {
+
+    //todo assert lengths of all the mostFreq* are same
+
+    //for each event mention, get the sentence id, and map it to a case class called contextDetails, which will have all of mostFreq* information
+    //note: zipping through only the list of one mostFreq* since all of them should have same lenghts.
+    for ((mostFreq, i) <- mostFreqLocation0Sent.zipWithIndex) {
+      checkSentIdContextDetails(sentidContext,mostFreq.sentId,
+        ContextDetails(mostFreq.mention,
+          checkIfEmpty(mostFreq.mostFreqEntity),
+          checkIfEmpty(mostFreqLocation1Sent(i).mostFreqEntity),
+          checkIfEmpty(mostFreqLocationOverall(i).mostFreqEntity),
+          checkIfEmpty(mostFreqDate0Sent(i).mostFreqEntity),
+          checkIfEmpty(mostFreqDate1Sent(i).mostFreqEntity),
+          checkIfEmpty(mostFreqDateOverall(i).mostFreqEntity),
+          checkIfEmpty(mostFreqCrop0Sent(i).mostFreqEntity),
+          checkIfEmpty(mostFreqCrop1Sent(i).mostFreqEntity),
+          checkIfEmpty(mostFreqCropOverall(i).mostFreqEntity),
+          checkIfEmpty(mostFreqFertilizer0Sent(i).mostFreqEntity),
+          checkIfEmpty(mostFreqFertilizer1Sent(i).mostFreqEntity),
+          checkIfEmpty(mostFreqFertilizerOverall(i).mostFreqEntity)
+        )
+      )
+    }
+  }
+  //if none, return "N/A", else return String value of entity e.g.:"SENEGAL"
+  def checkIfEmpty(mostFreqEntity: Option[String]): String = mostFreqEntity.getOrElse("N/A")
+
+  def findClosestNextLocation(mention: Mention, locations: Seq[Mention]): Mention = {
+    if (locations.length == 1) return locations.head
+    val nextLocations = locations.filter(_.tokenInterval.start > mention.arguments("value").head.tokenInterval.start)
+    if (nextLocations.nonEmpty) nextLocations.minBy(_.tokenInterval)
+    else null
+  }
+
+  def findClosestDate(mention: Mention, dates: Seq[Mention]): Mention = {
+    var minDist = 100
+    var minDistDate = dates.head
+    for (date <- dates) {
+      if (math.abs(mention.tokenInterval.start - date.tokenInterval.end ) < minDist | math.abs(mention.tokenInterval.end - date.tokenInterval.start) < minDist) minDistDate = date
```

Comment:
|| ?

Reply:
Would dates.minBy work here?  That 100 might lead to an incorrect answer, I think.